src/gui/widgets/menu_button.hpp

```diff
@@ -95,6 +95,17 @@ class menu_button : public styled_widget, public selectable_item
 		return values_[selected_]["label"];
 	}
 
+	/** Selects the row whose label equals the given string */
+	void set_selected_from_string(std::string label, bool fire_event = false)
```

Comment:
This isn't used, even with the unit editing dialog.

It's not a reliable assumption that there will only be one such row, so this relies on the design of the code using the menu_button. I think it should be left to that code, and this method should be removed.

Reply:
If this is removed, there must at least be a way to search for a specific string in the list of entries. Iterating over them is good enough. Maybe that already exists? A find function would also work.

Reply:
The dialog that owns the menu_button can already search the data that was used to populate the menu_button, so why does it need to search the displayable strings?

Reply:
> The dialog that owns the menu_button can already search the data that was used to populate the menu_button

Then all is well.

Reply:
Clarification on this : This was added as a convenience function that saves anybody using this from actually writing the same for loops over and over. Often we have a string and not the index to set in the `menu_button`. It simplified the unit_editor code a lot and saved some debug time.

Reply:
If we remove this, then we also need some kind of function in `utils` most probably that searches through a `std::vector<std::string>` or similar.

Reply:
?

Is `std::find` insufficient?

Reply:
Removed. Thanks for `std:find`, didn't knew about that.